server/api/extension-hooks/routes/put_update.js

```diff
@@ -0,0 +1,35 @@
+import _ from 'lodash';
+import compileRule from '../../../lib/compileRule';
+
+module.exports = (server) => ({
+  method: 'PUT',
+  path: '/.extensions/on-update',
+  config: {
+    auth: false,
+    pre: [
+      server.handlers.validateHookToken('/meta'),
+      server.handlers.managementClient
+    ]
+  },
+  handler: (req, reply) => {
+    req.pre.auth0
+      .rules
+      .getAll()
+      .then(rules => req.storage.getConfiguration()
+        .then((config) => {
+          const payload = {
+            name: 'auth0-authorization-extension',
+            script: compileRule(config, 'auth0-authz-extension')
```

Comment:
We should not use the config here. In v1 the behavior was fixed: add groups to token and persist them in app metadata. So for migration purposes, that's how we should update the rule.